applications/ShapeOptimizationApplication/python_scripts/algorithm_bead_optimization.py

```diff
@@ -46,20 +45,27 @@ def __init__(self, optimization_settings, analyzer, communicator, model_part_con
                 "line_search_type"           : "manual_stepping",
                 "normalize_search_direction" : true,
                 "step_size"                  : 1.0
-            }
+            },
+            "filter_penalty_term"           : false,
+            "penalty_filter_radius"         : -1.0
         }""")
         self.algorithm_settings =  optimization_settings["optimization_algorithm"]
         self.algorithm_settings.RecursivelyValidateAndAssignDefaults(default_algorithm_settings)
 
         self.optimization_settings = optimization_settings
         self.mapper_settings = optimization_settings["design_variables"]["filter"]
 
+        # change the default value of penalty filtering to the filter radius of the mapper.
+        if self.algorithm_settings["penalty_filter_radius"].GetDouble() == -1.0:
```

Comment:
Hm this is not nice
How about setting this as a string => „automatic“ and then replace it in this case?

Same is done for the „End“ in the interval of the processes

Reply:
Thought about this before, but the difference here is that i wan't to assign a value from the same settings object in the automatic case. In order to not break the validation, i need to to that before the Validate call and that means i need to get the default value from the settings before validating it...

Reply:
Hmm thinking a bit about it: What about making the penalty_filter_radius a necessary option. Meaning it always has to be specified, and if the penalty filter should have the same radius as the regular one, then one would also need to specify it. Internally then, one could, in an if condition, ask if both radii match to avoid creating an extra penalty filter.

So basically following the basic python idea of "explicit" over "implicit".

Reply:
:+1: